src/attributes/classes.js

```diff
@@ -23,9 +39,7 @@ jQuery.fn.extend( {
 			} );
 		}
 
-		if ( typeof value === "string" && value ) {
-			classes = value.match( rnothtmlwhite ) || [];
```

Comment:
It has to be backwards compatible but that is still possible without duplicating the code.